builder/amazon/common/block_device.go

```diff
@@ -42,7 +42,7 @@ func buildBlockDevices(b []BlockDevice) []*ec2.BlockDeviceMapping {
 		// You cannot specify Encrypted if you specify a Snapshot ID
 		if blockDevice.SnapshotId != "" {
 			ebsBlockDevice.SnapshotID = &blockDevice.SnapshotId
-		} else {
+		} else if blockDevice.Encrypted {
```

Comment:
Found an issue where if you override some settings of the built-in device, but not all of them, this could get set and we'd get an error in building